src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,202 @@
+        """Returns the state of the connection with the given network id."""
+        state = NM_CONNECTION_DISCONNECTED
+
+        active_connection = self._get_active_connection(network_id)
+
+        if active_connection is None:
+            self.logger.warning('Connection disconnected.')
+
+        else:
+            if hasattr(active_connection, 'State'):
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATING:
+                    state = NM_CONNECTION_ACTIVATING
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATED:
+                    state = NM_CONNECTION_ACTIVE
+
+        return state
+
+    def _get_device_objects(self, wired_interface_name, wifi_interface_name):
+        """Store references to the device objects that are used frequently."""
+
+        for device in NetworkManager.NetworkManager.GetDevices():
+            if device.Interface == wired_interface_name:
+                self.wired_device = device
+
+            if device.Interface == wifi_interface_name:
+                self.wireless_device = device
+
+        if self.wired_device is None:
+            raise DeviceNotFoundException('Defined wired device %s was not found.' %
+                wired_interface_name)
+
+        if self.wireless_device is None:
+            raise DeviceNotFoundException('Defined wireless device %s was not found.' %
+                wifi_interface_name)
+
+    def _get_device_for_connection(self, connection):
+        """Get the device object a connection object needs to connect with."""
+
+        network_device = self.wireless_device
+
+        connection_id = connection.GetSettings()['connection']['id']
+        if connection_id == self.wired_network_name:
+            network_device = self.wired_device
+
+        return network_device
+
+    def _wait_for_connection(self, connection):
+        """Wait timeout number of seconds for an active connection to be ready.
+        return True if it connects within timeout, False otherwise.
+        """
+        success = False
+        give_up = False
+        time_to_give_up = time.time() + self.network_activation_timeout
+        network_id = connection.GetSettings()['connection']['id']
+
+        self.logger.debug('Waiting for connection %s...' % network_id)
+        while (success is False and give_up is False):
+            time.sleep(NETWORKMANAGER_ACTIVATION_CHECK_DELAY)
+
+            connection_state = self._get_connection_state(network_id)
+
+            if connection_state is NM_CONNECTION_ACTIVE:
+                self.logger.debug('Connection successful.')
+                success = True
+
+            elif connection_state is NM_CONNECTION_DISCONNECTED:
+                self.logger.warning('Connection disconnected, giving up.')
+                give_up = True
+
+            elif time.time() > time_to_give_up:
+                self.logger.warning('Connection timed out, giving up.')
```

Comment:
This and the prior warning should probably be broken up into two sentences. 'Connection timed out. Giving up.'

Also, it should specify which connection timed out.